File: Data Structures & Algorithms/anagram-groups/submission-1.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = defaultdict(list)

        for s in strs:
            ref = [0] * 26
            for c in s:
                ref[ord(c) - ord("a")] += 1
            res[tuple(ref)].append(s)
        
        return res.values()

File: Data Structures & Algorithms/longest-palindromic-substring/submission-1.cs
public class Solution {
    public string LongestPalindrome(string s) {
        string res = "";

        for (int i = 0; i < s.Length; i++) {
            int l = i;
            int r = i;
            // odd
            while (l >= 0 && r < s.Length && s[l] == s[r]) {
                if ((r - l + 1) > res.Length) {
                    res = s[l..(r+1)];
                }

                l--;
                r++;
            }

            l = i;
            r = i + 1;
            // even
            while (l >= 0 && r < s.Length && s[l] == s[r]) {
                if ((r - l + 1) > res.Length) {
                    res = s[l..(r+1)];
                }
                l--;
                r++;
            }
        }

        return res;        
    }
}
